src/interfaces/graphql/resolvers/admin/Query.ts

```diff
@@ -12,5 +14,15 @@ export const Query = {
       });
     }
     return { __typename: 'RefreshTokenPayload', success: true };
+  },
+  pageBlocks: async (_: unknown, args: { slug: string }) => {
+    const service = PageService(PageRepository());
+    const page = await service.getPage(args.slug);
+
+    if (!page) {
+      throw new GraphQLError('Page not found', { extensions: { code: 'NOT_FOUND' } });
+    }
+
+    return page;
```

Comment:
```suggestion
  pageBlocks: async (_: unknown, args: { slug: string }) => {
    if (!context.admin) {
      throw new GraphQLError('You must be logged in to access this resource.', {
        extensions: {
          code: 'UNAUTHENTICATED'
        }
      });
    }
    const service = PageService(PageRepository());
    const page = await service.getPage(args.slug);

    if (!page) {
      throw new GraphQLError('Page not found', { extensions: { code: 'NOT_FOUND' } });
    }

    return page;
```

add it here please to handle UNAUTHENTICATED admins